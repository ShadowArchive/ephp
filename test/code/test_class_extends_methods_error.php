<?php

class A {
    private function a1() {
        print "calling to protected " . __CLASS__ . " -> a1()\n";
    }

    public function a2() {
        print "calling to public " . __CLASS__ . " -> a2()\n";
    }
}

class B extends A {
    private function b1() {
        print "calling to protected " . __CLASS__ . " -> b1() who is ";
        $this->a1();
    }

    public function b2() {
        print "calling to public " . __CLASS__ . " -> b2() who is ";
        $this->a2();
    }

    public function calling() {
        $this->b1();
        $this->b2();
    }
}

$b = new B;
$b->calling();

