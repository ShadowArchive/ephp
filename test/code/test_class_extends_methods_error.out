calling to protected B -> b1() who is 
Fatal error: Call to private method A::a1() from context 'B' in {{CWD}}/test/code/test_class_extends_methods_error.php on line 16
